### PR TITLE
Support `quilt.mod.json` metadata

### LINKED ISSUE #428: Quilt: parse quilt.mod.json
### Role

i use polymc and i plan ditching fabric for quilt

### Suggestion

currently, we just parse fabric.mod.json (fabric) and the forge thingy

### Benefit

parsing quilt mod infos

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

https://github.com/QuiltMC/rfcs/blob/master/specification/0002-quilt.mod.json.md

### PR DESCRIPTION
**Pull request by Scrumplex**
Closes #428 

Hopefully I didn't miss anything

### CODE REVIEW

**Review by flowln — APPROVED** [review]
LGTM!

It's not required, but maybe we could start working towards #275 by parsing [the `icon` field](https://github.com/QuiltMC/rfcs/blob/master/specification/0002-quilt.mod.json.md#the-icon-field) and storing a path in a field in `ModDetails`!

File: launcher/minecraft/mod/LocalModParseTask.cpp [hunk 1]
```diff
@@ -262,6 +263,43 @@ std::shared_ptr<ModDetails> ReadFabricModInfo(QByteArray contents)
     return details;
 }
 
+// https://github.com/QuiltMC/rfcs/blob/master/specification/0002-quilt.mod.json.md#the-schema_version-field
```
Comment (TheGlitch76) [inline launcher/minecraft/mod/LocalModParseTask.cpp]: Would be wise to point to the specific commit that you are implementing the spec from

Comment (flowln) [reply launcher/minecraft/mod/LocalModParseTask.cpp]: If the intent is updating the spec version with each new revision, then I think it's fine leaving this link as it is, and add a link to the specific commit above the `if (schemaVersion == 1)` line.

**Review by ZekeZDev — APPROVED** [review]

**Review by txtsd — APPROVED** [review]

**Comment by txtsd** [comment]
Are you going to parse the icon field in this PR or later?

**Comment by Scrumplex** [comment]
> Are you going to parse the icon field in this PR or later?

Different PR

**Pull request merged**